src/modules/commander/Commander.cpp

```diff
@@ -2123,7 +2173,8 @@ void Commander::handleAutoDisarm()
 			if (_actuator_armed.manual_lockdown) {
 				disarm(arm_disarm_reason_t::kill_switch, true);
 
-			} else {
+			} else if (!_param_com_throw_en.get()) {  // don't disarm if throw
```

Comment:
maybe rather do the skip where it's also done for HITL a couple of lines above

Reply:
Good catch! It also motivated me to add a helper function `isThrowLaunchActive` to do the check instead of checking for appropriate states each time. I've pushed it with the other suggested changes.